services/authentication.go

```diff
@@ -0,0 +1,79 @@
+package services
+
+import (
+	"context"
+	"encoding/base64"
+	"net/http"
+	"strings"
+
+	"github.com/kacperf531/sockchat/api"
+)
+
+type SockchatAuthService struct {
+	UserProfiles api.SockchatProfileStore
+}
+
+type authWrapper struct {
+	Username string
+	Password string
+}
+
+func (s *SockchatAuthService) AuthenticateFromBasicToken(ctx context.Context, token string) (bool, error) {
+	if !isBasicToken(token) {
+		return false, api.ErrBasicTokenRequired
+	}
+	auth, err := decodeToken(token)
+	if err != nil {
+		return false, err
+	}
+	return s.UserProfiles.IsAuthValid(ctx, auth.Username, auth.Password), nil
+}
+
+func isBasicToken(token string) bool {
+	return strings.HasPrefix(token, "Basic ")
+}
+
+func decodeToken(token string) (*authWrapper, error) {
+	encoded := strings.TrimPrefix(token, "Basic ")
```

Comment:
If `token` doesn't start with prefix `Basic `, it is passed to `encoded` unchanged. Is this the desired behavior and base64 encoded `{username}:{password}` should be accepted as well?

If you want to require `Basic ` in the prefix you can use the [strings.CutPrefix](https://pkg.go.dev/strings#CutPrefix) like in the suggestion below or split the `token` by space and validate first part.
```suggestion
	encoded, foundBasic := strings.CutPrefix(token, "Basic ")
	if !foundBasic {
		return nil, api.ErrBasicTokenRequired
	}
```